.github/workflows/python-workflow.yml

```diff
@@ -15,74 +15,42 @@ on:
     branches:
       - master
       - 'v1.*'
-  pull_request:
+    paths:
+      - '**.py'
+  pull_request_target:
```

Comment:
This will not take the code from the PR branch, but from the branch the PR is based on. I think we want to lint the code from the PR branch, how do we achieve that in this case?

Reply:
Do we need the `pull_request_target` trigger or can we keep `pull_request`? If not then we can also force `actions/checkout` to use the PR branch. But I have never tested this.

Reply:
The `comment-on-pr` step introduced in this PR doesn't work when triggered on a PR as it doesn't have the required permissions.

Reply:
Damn, you are completely right. I missed this difference to the other refactored check in #14474 .

I guess conceptually you will still need the workaround with artifacts and a `workflow_dispatch` trigger. However, I guess I am currently not able to adopt this PR to that big of a change.